crates/pixi_command_dispatcher/src/source_build/mod.rs

```diff
@@ -116,34 +128,110 @@ impl SourceBuildSpec {
             .await
             .map_err_with(SourceBuildError::Initialize)?;
 
-        if backend.capabilities().provides_conda_build_v1() {
-            let built_package = self.build_v1(command_dispatcher, backend).await?;
+        // Determine the working directory for the build.
+        let working_directory = match std::mem::take(&mut self.working_directory) {
+            Some(working_directory) => working_directory,
+            None => command_dispatcher.cache_dirs().working_dirs().join(
+                WorkDirKey {
+                    source: SourceRecordOrCheckout::Record {
+                        pinned: self.source.clone(),
+                        package_name: self.package.name.clone(),
+                    },
+                    host_platform: self.build_environment.host_platform,
+                    build_backend: backend.identifier().to_string(),
+                }
+                .key(),
+            ),
+        };
+
+        // Clean the working directory if requested.
+        if self.clean {
+            if let Err(err) = fs_err::remove_dir_all(&working_directory) {
+                return Err(CommandDispatcherError::Failed(
+                    SourceBuildError::CleanWorkingDirectory(working_directory, err),
+                ));
+            }
```

Comment:
Why not `.map_err`?

Reply:
So I can move the `working_directory`. Using `map_err` I have to clone it because its also used in the code below.